### ISSUE TITLE
Should entropy of different actions be reduced by sum?

### ISSUE THREAD

**Issue by bigtreeliujie** [reinforceio/tensorforce#177]
Entropy is used in some PG algorithms (such as PPO, TRPO). The entropy of different actions is now reduced by mean:
```Python
entropy = tf.reduce_mean(input_tensor=tf.concat(values=entropies, axis=1), axis=1)
```
In my option, since the action number may be different in actions, the max values of entropy of each action are different, thus the more number of actions makes the larger max values. As a result, the mean of entropy of different actions do not make sense. If we do the mean,  we give a "weight" to the entropy by the number of actions, the action with more number has a larger weight. So, we should use reduce_sum to merge the entropy from different actions, and "sum" means the total information entropy of all the actions.

By the way, it's OK to reduce the entropy from different batch record, which is exactly the way in current implementation.

**Comment by AlexKuhnle** [comment]
Would you suggest to apply `tf.reduce_mean` per action, but `tf_reduce_sum` for the resulting action means, or just replace `tf.reduce_mean` with `tf.reduce_sum` in the line you refer to? I assume the former, since it seems to me the latter wouldn't effectively change much. The former kind of makes sense to me, one would interpret an action consisting of potentially multiple sub-actions, but these sub-actions are not 'on the same level' as the actions  in terms of contribution to the loss. Would you agree with this understanding?

**Comment by bigtreeliujie** [comment]
Yes, I agree with you. But, I think this is depends on the user, and current  implementation is OK. So, I close this issue.

**Comment by AlexKuhnle** [comment]
Hmm, so you think it is fine after all? It won't necessarily make a massive difference, but I see the point of the latter possibility...
